cheddar/cheddar-integration-mocks/src/main/java/com/clicktravel/infrastructure/persistence/inmemory/database/InMemoryDatabaseTemplate.java

```diff
@@ -163,6 +175,28 @@ private void deleteUniqueConstraints(final Item item) {
         }
     }
 
+    /**
+     * Checks the presence of a matching unique constraint for the given item property
+     * 
+     * @param item The Item for which to check unique constraints
+     * @param propertyName The name of the property for which the unique constraint should be checked
+     * @param propertyValue The value of the property which needs to be checked for presence of unique constraint
+     * @return
+     */
+    public boolean hasMatchingUniqueConstraint(final Item item, final String propertyName, final String propertyValue) {
```

Comment:
Is this method only used in unit tests? If so, is it possible to move it to the unit test source code tree?